machine/v1/types_controlplanemachineset.go

```diff
@@ -248,6 +249,10 @@ type FailureDomains struct {
 	// GCP configures failure domain information for the GCP platform.
 	// +optional
 	GCP *[]GCPFailureDomain `json:"gcp,omitempty"`
+
+	// OpenStack configures failure domain information for the OpenStack platform.
+	// +optional
+	OpenStack *[]OpenStackFailureDomain `json:"openstack,omitempty"`
```

Comment:
While this goes against the other items in this list, do you think you could change this to not be a pointer, and see if that still works? I think that would have made our zero length failure domains list bug much more obvious. I need to double check the unions guidance but I think that should be ok, the idea is that it needs to be omitted when empty, so we can change nil checks for length checks

Reply:
Sounds good. I'm testing it now.
I'll update the other PRs accordingly.